### ISSUE TITLE
Transpiled webpack bundle does not export hyphenated package name via require

### ISSUE THREAD

**Issue by the4dpatrick** [chentsulin/electron-react-boilerplate#232]
[StackOverflow question](http://stackoverflow.com/questions/37505105/transpiled-webpack-bundle-does-not-export-hyphenated-package-name-via-require)
[Example Repo](https://github.com/the4dpatrick/congenial-barnacle)
[Breaking commit](https://github.com/the4dpatrick/congenial-barnacle/commit/112cebf4df46e4683cc01fcb80e6df253a9924ce)

The `Main.js` file within the app's contents, after running `npm run package`, includes a line `function(e,t){e.exports=fluent-ffmpeg}`. When I try to open the app, I receive this error: `Uncaught Exception: ReferenceError: fluent is not defined`

I can fix this by manually editing the aforementioned line to read: `function(e,t){e.exports=require("fluent-ffmpeg")}`

**Steps to reproduce**
* I cloned this repo
* added the `fluent-ffmpeg` package
* add `import ffmpeg from 'fluent-ffmpeg'` into `main.development.js`
* `npm run package`
* open app

[My Example Repo](https://github.com/the4dpatrick/congenial-barnacle) shows this same error. 

Any help with making `fluent-ffmpeg` work with this repo would be appreciated.

**Comment by eugenehp** [comment]
@the4dpatrick can you please explain what exactly you are trying to achieve with including the fluent ffmpeg package in your frontend?

**Comment by the4dpatrick** [comment]
The goal of including `fluent-ffmpeg` is to create a png from a single frame of a video. @eugenehp 
My understanding is that these types of packages should run on the main process vs the renderer process. Is `main.development.js` not the right place for that?

**Comment by eugenehp** [comment]
@the4dpatrick see my suggestion in my recent commit https://github.com/eugenehp/congenial-barnacle

**Comment by the4dpatrick** [comment]
@eugenehp I can see where this can work after running the `npm run dev` command where the server is spun up for development, but after I want to package the application, I would need a separate API running to access ffmpeg. I want to be able to run `fluent-ffmpeg` apart of an offline application without spinning up a server.

I might have found a fix by changing `externals` inside the `webpack.config.base.js` to read:

```
externals: {
    'fluent-ffmpeg': 'fluent-ffmpeg'
  }
```

I have yet to confirm this actually works

**Comment by eugenehp** [comment]
@the4dpatrick interesting. how do you plan to build a standalone application once you get reed of the dev server? I'd say we can configure something like this inside that packager that you are going to use to deliver the production grade app.

**Comment by FraGoTe** [comment]
Webpack recommend to declare that way so I think you should send a PR to boiler library to modify the default from array to object in external declaration.

PD . https://webpack.github.io/docs/library-and-externals.html

**Comment by the4dpatrick** [comment]
I was able to [solve the issue](https://github.com/the4dpatrick/congenial-barnacle/commit/b2618f0db9d97412571c21b37c7542f0795e7fba) by simply setting the `output.libraryTarget` within `webpack.config.electron.js` file. 
```
output: {
    path: path.join(__dirname, 'dist'),
    filename: 'bundle.js',
    libraryTarget: 'commonjs2'
  },
```

@eugenehp with this boilerplate we could run `npm run package` to deliver the production app.
@FraGoTe I was able to maintain `externals` being an array. The only possible change to this boilerplate that is related to this issue might be the setting `output.libraryTarget`
